src/verilog/aval_bval_encoding.cpp

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************\
+
+Module: aval/bval encoding
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+#include "aval_bval_encoding.h"
+
+#include <util/arith_tools.h>
+#include <util/bitvector_expr.h>
+#include <util/bitvector_types.h>
+#include <util/mathematical_types.h>
+#include <util/std_expr.h>
+
+bv_typet aval_bval_type(std::size_t width, irep_idt source_type)
+{
+  PRECONDITION(!source_type.empty());
+  auto result = bv_typet{width * 2};
+  result.set(ID_C_verilog_aval_bval, source_type);
+  return result;
+}
+
+bv_typet lower_to_aval_bval(const typet &src)
```

Comment:
Shouldn't this be a `bitvector_typet` instead of just `typet`? This will then also avoid the `to_bitvector_type` a few lines further down.